text/0016-standard-modes.md

```diff
@@ -0,0 +1,306 @@
+          The precise flight path and landing behaviour depend on vehicle configuration and type.
+        </description>
+      </entry>
+      <entry value="5" name="MAV_STANDARD_MODE_SAFE_RECOVERY">
+        <description>Safe recovery mode (auto).
+          Automatic mode that takes vehicle to a predefined safe location via a safe flight path (rally point or mission defined landing) .
+          It may also automatically land the vehicle.
+          The precise return location, flight path, and landing behaviour depend on vehicle configuration and type.
+        </description>
+      </entry>
+      <entry value="6" name="MAV_STANDARD_MODE_MISSION">
+        <description>Mission mode (automatic).
+          Automatic mode that executes MAVLink missions.
+          Missions are executed from the current waypoint as soon as the mode is enabled.
+        </description>
+      </entry>
+      <entry value="7" name="MAV_STANDARD_MODE_LAND">
+        <description>Land mode (auto).
+          Automatic mode that lands the vehicle at the current location.
+          The precise landing behaviour depends on vehicle configuration and type.
+        </description>
+      </entry>
+      <entry value="8" name="MAV_STANDARD_MODE_TAKEOFF">
+        <description>Takeoff mode (auto).
+          Automatic takeoff mode.
+          The precise takeoff behaviour depends on vehicle configuration and type.
+        </description>
+      </entry>
+    </enum>
+```
+
+Notes:
+- In future this set may be extended with additional flight modes and component specific modes.
+- The "return home" and "safe recovery" modes both serve the same function: getting the vehicle to a safe location in the event of a failsafe or for landing.
+  They are separated because from the user perspective the behaviour is quite different and the GCS needs to be able to present that behaviour differently.
+  - The existance of these as separate modes means that the autopilot must have separate custom modes for these if we don't publish the standard mode independently. 
+
+
+## Setting Modes
+
+A standard mode should be set using a new command: `MAV_CMD_DO_SET_STANDARD_MODE`.
+
+```xml
+      <entry value="262" name="MAV_CMD_DO_SET_STANDARD_MODE" hasLocation="false" isDestination="false">
+        <description>Enable the specified standard MAVLink mode.
+          If the mode is not supported the vehicle should ACK with MAV_RESULT_FAILED.
+        </description>
+        <param index="1" label="Standard Mode" enum="MAV_STANDARD_MODE">The mode to set.</param>
+        <param index="2" reserved="true" default="0"/>
+        <param index="3" reserved="true" default="0"/>
+        <param index="4" reserved="true" default="0"/>
+        <param index="5" reserved="true" default="0"/>
+        <param index="6" reserved="true" default="0"/>
+        <param index="7" reserved="true" default="NaN"/>
+      </entry>
+```
+
+
+## Getting Available Modes
```

Comment:
I'm wondering whether using a metadata json file would be the better option here.

advantages:
- can be extended easily with any kind of metadata, such as whether a mode is advanced or not, group into manual vs autonomous, etc.
- translation
- avoid another protocol for specific metadata

potential disadvantages:
- less dynamic
  - modes for all vehicle types can be included in the same file, and be flagged as type-specific
  - configuration-dependent modes could be handled via conditions (similar to https://github.com/mavlink/mavlink/blob/master/component_information/actuators.schema.json#L8)
- a higher-level protocol is required (ftp)

This is similar to the list of supported commands.

Reply:
IMO component information would be a perfect fit for this. In particular it would remove the main limitation of the current version, which is that custom mode strings can't be translated.

However component information is not yet "standard". I'd happily have a parallel path for it to be fetched in this way, but  not "instead of".

Reply:
Added as an alternative. I'm going to mark this as Resolved. I hope we get to do it.